File: app/src/main/java/com/mytlogos/enterprise/background/api/model/ClientExternalMediaList.java
package com.mytlogos.enterprise.background.api.model;

import java.util.Arrays;

/**
 * API Model for ExternalList.
 */
public class ClientExternalMediaList {
    private final String uuid;
    private final int id;
    private final String name;
    private final int medium;
    private final String url;
    private final int[] items;

    public ClientExternalMediaList(String uuid, int id, String name, int medium, String url,
                                   int[] items) {
        this.uuid = uuid;
        this.id = id;
        this.name = name;
        this.medium = medium;
        this.url = url;
        this.items = items;
    }

    public String getUuid() {
        return uuid;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public int getMedium() {
        return medium;
    }

    public String getUrl() {
        return url;
    }

    public int[] getItems() {
        return items;
    }

    @Override
    public String toString() {
        return "ClientExternalMediaList{" +
                "uuid='" + uuid + '\'' +
                ", id=" + id +
                ", name='" + name + '\'' +
                ", medium=" + medium +
                ", url='" + url + '\'' +
                ", items=" + Arrays.toString(items) +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ClientExternalMediaList that = (ClientExternalMediaList) o;

        return getId() == that.getId();
    }

    @Override
    public int hashCode() {
        return getId();
    }
}

File: app/src/main/java/com/mytlogos/enterprise/background/api/model/ClientExternalMediaList.kt
package com.mytlogos.enterprise.background.api.model

/**
 * API Model for ExternalList.
 */
data class ClientExternalMediaList(
    val uuid: String,
    val id: Int,
    val name: String,
    val medium: Int,
    val url: String,
    val items: IntArray,
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as ClientExternalMediaList
        return id == that.id
    }

    override fun hashCode(): Int {
        return id
    }
}